ENIGMAsystem/SHELL/Universal_System/Extensions/DataStructures/data_structures.cpp

```diff
@@ -52,6 +54,16 @@ template<typename t> bool tequal(t v1, t v2) { return v1 == v2; }
 template<> bool tequal(float v1, float v2)   { return fequal(v1, v2); }
 template<> bool tequal(double v1, double v2) { return fequal(v1, v2); }
 
+namespace enigma_user {
+  // why the fuck can't we just include mathnc.h here?
+  // for connecting ds_list_shuffle and ds_grid_shuffle
+  // to randomize as specified by GM docs
+  extern ma_scalar random(ma_scalar n);
+  static inline int random_integer(int x) {
```

Comment:
Doesn't this function exist? Anyway, don't declare this in `enigma_user`, even if it's static.

Reply:
Yes, HOWEVER:

1) It's inclusive which doesn't work with random_shuffle:
https://github.com/enigma-dev/enigma-dev/blob/edccdb82110f545ec7051e48b9df80ce1a2c5c50/ENIGMAsystem/SHELL/Universal_System/mathnc.h#L220
2) I am not allowed to include `mathnc.h`:
https://github.com/enigma-dev/enigma-dev/blob/edccdb82110f545ec7051e48b9df80ce1a2c5c50/ENIGMAsystem/SHELL/Universal_System/mathnc.h#L41

Reply:
I see... that is kind of sad. But all the function does is cast the result of `random`, so why not just cast the result of random.

Also, feel free to extract the `random` family of functions into a separate header. I forbade including `mathnc` outside of the SHELLmain because polygone kept using it for GM's math functions instead of just using the C++ equivalents.

Reply:
Alright, I've tweaked it: 45089f939b72cfd70ca6490b53eff47771afa7fe

I moved all the random stuff to `Universal_System/mathrand.h` which allows me to include elsewhere. I also used the random generator for `random_shuffle` to namespace enigma and out of enigma_user.

It appears to continue working and shuffling the list correctly.